### PR TITLE
fix(asm): capture XML parsing errors on body parse

### PR DESCRIPTION
**Pull request by juanjux**
Signed-off-by: Juanjo Alvarez <juanjo.alvarezmartinez@datadoghq.com>

## Description
xmltodict could raise some exceptions that we weren't catching on XML body parsing.

## Checklist
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by gnufede — DISMISSED** [review]

File: ddtrace/contrib/django/utils.py [hunk 1]
```diff
@@ -249,6 +249,7 @@ def _before_request_tags(pin, span, request):
 
 
 def _extract_body(request):
+    print("XXX extract_body")
```
Comment (brettlangdon) [inline ddtrace/contrib/django/utils.py:252]:
```suggestion
```

File: releasenotes/notes/capture-xml-parsing-errors-e6c8c761ed026ce3.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    ASM: capture more potential XML parsing exceptions on body parse.
```
Comment (brettlangdon) [inline releasenotes/notes/capture-xml-parsing-errors-e6c8c761ed026ce3.yaml:4]:
```suggestion
    ASM: Do not raise exceptions when failing to parse XML request body.
```

File: tests/contrib/flask/test_flask_appsec.py [hunk 3]
```diff
@@ -235,3 +235,21 @@ def test_flask_body_json_empty_body_logs_warning(self):
...
+        with self._caplog.at_level(logging.WARNING), override_global_config(dict(_appsec_enabled=True)):
+            self._aux_appsec_prepare_tracer()
+            self.client.post("/", data="bad xml", content_type="application/xml")
+            assert "Failed to parse werkzeug request body" in self._caplog.text
+
+    def test_flask_body_xml_empty_logs_warning(self):
+        with self._caplog.at_level(logging.WARNING), override_global_config(dict(_appsec_enabled=True)):
+            self._aux_appsec_prepare_tracer()
+            self.client.post("/", data="bad xml", content_type="application/xml")
+            assert "Failed to parse werkzeug request body" in self._caplog.text
```
Comment (brettlangdon) [inline tests/contrib/flask/test_flask_appsec.py:255]: What is the difference between these two? they look identical to me

Comment (juanjux) [reply tests/contrib/flask/test_flask_appsec.py]: Good catch, second one should have an empty strings, fixed.

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Comment by juanjux** [comment]
@brettlangdon all CI working except the usual flaky windows build stuff (did a lot of retries), PTAL when you can and maybe manual merge.

**Comment by juanjux** [comment]
@mergifyio backport 1.6

**Comment by juanjux** [comment]
@mergifyio backport 1.5

**Pull request merged**